plugins/modules/postgresql_script.py

```diff
@@ -0,0 +1,398 @@
+  link: https://www.postgresql.org/docs/current/ddl-schemas.html
+
+author:
+- Douglas J Hunley (@hunleyd)
+- A. Hart (@jtelcontar)
+- Daniel Scharon (@DanScharon)
+- Andrew Klychkov (@Andersson007)
+
+extends_documentation_fragment:
+- community.postgresql.postgres
+
+notes:
+- Does not support C(check_mode).
+'''
+
+EXAMPLES = r'''
+# Assuming that the file contains
+# SELECT * FROM id_talbe WHERE id = %s,
+# '%s' will be substituted with 1
+- name: Run query from SQL script using UTF-8 client encoding for session and positional args
+  community.postgresql.postgresql_script:
+    db: test_db
+    path: /var/lib/pgsql/test.sql
+    positional_args:
+      - 1
+    encoding: UTF-8
+
+# Assuming that the file contains
+# SELECT * FROM test WHERE id = %(id_val)s AND story = %(story_val)s,
+# %-values will be substituted with 1 and 'test'
+- name: Select query to test_db with named_args
+  community.postgresql.postgresql_script:
+    db: test_db
+    path: /var/lib/pgsql/test.sql
+    named_args:
+      id_val: 1
+      story_val: test
+
+- block:
+  # Assuming that the the file contains
+  # SELECT * FROM test_array_table WHERE arr_col1 = %s AND arr_col2 = %s
+  # Pass list and string vars as positional_args
+  - name: Set vars
+    ansible.builtin.set_fact:
+      my_list:
+      - 1
+      - 2
+      - 3
+      my_arr: '{1, 2, 3}'
+  - name: Passing positional_args as arrays
+    community.postgresql.postgresql_script:
+      path: /var/lib/pgsql/test.sql
+      positional_args:
+        - '{{ my_list }}'
+        - '{{ my_arr|string }}'
+
+# Assuming that the the file contains
+# SELECT * FROM test_table,
+# look into app1 schema first, then,
+# if the schema doesn't exist or the table hasn't been found there,
+# try to find it in the schema public
+- name: Select from test using search_path
+  community.postgresql.postgresql_script:
+    path: /var/lib/pgsql/test.sql
+    search_path:
+    - app1
+    - public
+
+# If you use a variable in positional_args/named_args that can
+# be undefined and you wish to set it as NULL, the constructions like
+# "{{ my_var if (my_var is defined) else none | default(none) }}"
+# will not work as expected substituting an empty string instead of NULL.
+# If possible, we suggest to use Ansible's DEFAULT_JINJA2_NATIVE configuration
+# (https://docs.ansible.com/ansible/latest/reference_appendices/config.html#default-jinja2-native).
+# Enabling it fixes this problem. If you cannot enable it, the following workaround
+# can be used.
+# You should precheck such a value and define it as NULL when undefined.
+# For example:
+- name: When undefined, set to NULL
+  set_fact:
+    my_var: NULL
+  when: my_var is undefined
+
+# Then, assuming that the file contains
+# INSERT INTO test_table (col1) VALUES (%s)
+- name: Insert a value using positional arguments
+  community.postgresql.postgresql_script:
+    path: /var/lib/pgsql/test.sql
+    positional_args:
+    - '{{ my_var }}'
```

Comment:
```suggestion
- block:
    # If you use a variable in positional_args/named_args that can
    # be undefined and you wish to set it as NULL, constructions like
    # "{{ my_var if (my_var is defined) else none | default(none) }}"
    # will not work as expected substituting an empty string instead of NULL.
    # If possible, we suggest using Ansible's DEFAULT_JINJA2_NATIVE configuration
    # (https://docs.ansible.com/ansible/latest/reference_appendices/config.html#default-jinja2-native).
    # Enabling it fixes this problem. If you cannot enable it, the following workaround
    # can be used.
    # You should precheck such a value and define it as NULL when undefined.
    # For example:
    - name: When undefined, set to NULL
      set_fact:
        my_var: NULL
      when: my_var is undefined
    # Then, assuming that the file contains
    # INSERT INTO test_table (col1) VALUES (%s)
    - name: Insert a value using positional arguments
      community.postgresql.postgresql_script:
        path: /var/lib/pgsql/test.sql
        positional_args:
          - '{{ my_var }}'
```

Reply:
Looks much better, thanks! committed